tests/unit/test_simulated_stack.py

```diff
@@ -710,6 +724,89 @@ def testVariantAnnotationsSearch(self):
             fromJsonString(response.data)
         self.assertGreater(len(responseData.variantAnnotations), 0)
 
+    def testGetFeatureSet(self):
+        path = "/featuresets"
+        for dataset in self.dataRepo.getDatasets():
+            for featureSet in dataset.getFeatureSets():
+                responseObject = self.sendGetObject(
+                    path, featureSet.getId(), protocol.FeatureSet)
+                self.verifyFeatureSetsEqual(responseObject, featureSet)
+            for badId in self.getBadIds():
+                featureSet = features.AbstractFeatureSet(dataset, badId)
+                self.verifyGetMethodFails(path, featureSet.getId())
+        for badId in self.getBadIds():
+            self.verifyGetMethodFails(path, badId)
+
+    def testFeatureSetsSearch(self):
+        path = '/featuresets/search'
+        for dataset in self.dataRepo.getDatasets():
+            featureSets = dataset.getFeatureSets()
+            request = protocol.SearchFeatureSetsRequest()
+            request.datasetId = dataset.getId()
+            self.verifySearchMethod(
+                request, path, protocol.SearchFeatureSetsResponse, featureSets,
+                self.verifyFeatureSetsEqual)
+        for badId in self.getBadIds():
+            request = protocol.SearchFeatureSetsRequest()
+            request.datasetId = badId
+            self.verifySearchMethodFails(request, path)
+
+    def testGetFeature(self):
```

Comment:
Since these tests use the simulator they don't actually show the query interface at work. I'm not sure it makes sense to mock the entire query interface when we are backed by a database, but it will prove that protocol elements are well formed.